pkg/services/grafana-apiserver/rest/dualwriter.go

```diff
@@ -54,23 +58,31 @@ type LegacyStorage interface {
 type DualWriter struct {
 	Storage
 	legacy LegacyStorage
+	log    log.Logger
 }
 
 // NewDualWriter returns a new DualWriter.
 func NewDualWriter(legacy LegacyStorage, storage Storage) *DualWriter {
 	return &DualWriter{
 		Storage: storage,
 		legacy:  legacy,
+		log:     log.New("grafana-apiserver.dualwriter"),
 	}
 }
 
 // Create overrides the default behavior of the Storage and writes to both the LegacyStorage and Storage.
 func (d *DualWriter) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	if legacy, ok := d.legacy.(rest.Creater); ok {
-		_, err := legacy.Create(ctx, obj, createValidation, options)
+		created, err := legacy.Create(ctx, obj, createValidation, options)
 		if err != nil {
 			return nil, err
 		}
+		obj = created // write the updated version
+		rsp, err := d.Storage.Create(ctx, obj, createValidation, options)
+		if err != nil {
+			d.log.Error("unable to create object in duplicate storage", "error", err)
```

Comment:
could use klog -- but we are not yet doing that elsewhere, so i kept an explicit logger here